ql/src/java/org/apache/hadoop/hive/ql/exec/Utilities.java

```diff
@@ -1553,7 +1568,9 @@ public static void mvFileToFinalPath(Path specPath, Configuration hconf,
         Utilities.FILE_OP_LOGGER.trace("Moving tmp dir: {} to: {}", tmpPath, specPath);
         if(shouldAvoidRename(conf, hconf)){
           LOG.debug("Skipping rename/move files. Files to be kept are: " + filesKept.toString());
-          conf.getFilesToFetch().addAll(filesKept);
+          Set<FileStatus> statusFilesToKeep = new HashSet<>();
+          buildFileStatsList(statusList, filesKept, statusFilesToKeep);
```

Comment:
Instead of iterating filesKept and statusList again, better to get rid of filesKept altogether and just have statusKept in its place which is then populated in removeTempOrDuplicateFiles()

Reply:
Fixing